core/src/main/scala/io/snappydata/Literals.scala

```diff
@@ -269,6 +269,12 @@ object Property extends Enumeration {
         "as many partitions as executor cores clubbing multiple buckets " +
         "into each partition when possible.", Some(false), Constant.SPARK_PREFIX)
 
+  val ForcePrimariesInQueryRouting: SQLValue[Boolean] = SQLVal[Boolean](
+    s"${Constant.PROPERTY_PREFIX}forcePrimaries",
+    "Property to prefer using primary buckets in queries. This reduces " +
```

Comment:
Is this forcing or is this just a preference?
And too much of query routing term :-), may be we can use PreferPrimaryPartitions.

Reply:
Will change name